unity-client/Assets/Scripts/MainScripts/DCL/UserProfile/Helpers/UserProfileFetcher.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using DCL.Helpers;
+using DCL.Interface;
+
+public class UserProfileFetcher : IDisposable
+{
+    private readonly Dictionary<string, List<Promise<UserProfile>>> pendingPromises = new Dictionary<string, List<Promise<UserProfile>>>();
+    private bool isSubscribedToCatalog = false;
```

Comment:
We can avoid this flag by always unsubscribing before subscribing:
```
UserProfileController.userProfilesCatalog.OnAdded -= OnProfileAddedToCatalog;
UserProfileController.userProfilesCatalog.OnAdded += OnProfileAddedToCatalog;
```

This way we only subscribe once, regardless of how many times the code gets executed.

Reply:
great suggestion, I've changed it accordingly. thx!